### DISCUSSION TITLE
Publishing for Windows targeting both x86 and x64

### DISCUSSION THREAD

**Discussion by toptensoftware** [dotnet/runtime#60515] [Q&A]
What's the best way to publish a .NET Windows Desktop app that needs to be run in either x86 and x64 mode.

Background... the app needs to able to be run in either x86 or x64 by the user because it hosts third party plugins, many of which have never been updated to x64 and some users need to run the 32-bit version.  The app currently targets NET Framework 4.x and 99% of the app is built as "Any CPU" and there's two small bootup exe files - one for x86, one for x64.

Moving to .NET 5 I understand that publishing a .NET 5 application will automatically produce a .dll for the application with a boot .exe that loads the .net runtime before "running" the .dll.  ie: it automatically does something very similar to what I'm already manually doing.

My question are:

1. Do I need the entire published directory for each platform, or is it just the boot up .exe that's different such that I can trim things down and share everything else between the two platforms?
2. Can the publish tools, automatically produce a mixed mode (x86/64) published output?
3. Is there a better way to do this?

Any help or other suggestions appreciated.

**Comment by TonyValenti** [comment]
Following. I have this same situation and want to know the solution.
